lib/jekyll-admin/server/page.rb

```diff
@@ -12,6 +12,11 @@ class Server < Sinatra::Base
       end
 
       put "/*?/?:path.:ext" do
+        unless html_content?
+          status 422
+          content_type :json
+          return json "error_message" => "Invalid file extension for pages"
```

Comment:
Per the discussion above, not sure it's feasible to grab extensions that end up being html files. You can have many converters and they have a method `output_ext(ext)` which takes an ext. We can list all possible extensions